cmake/modules/extensions.cmake

```diff
@@ -2467,11 +2467,42 @@ function(zephyr_list transform list_var action)
 endfunction()
 
 # Usage:
-#   zephyr_get(<variable> [MERGE] [SYSBUILD [LOCAL|GLOBAL]])
+#   zephyr_var_name(<variable> <scope> <out>)
+#
+# Internal function for construction of scoped variable name expansion string.
+# Examples:
+#   reading a current scope FOO variable is identical to expand ${FOO}.
+#   reading a cache scope FOO variable is identical to expand $CACHE{FOO}.
+#
+# this functions will return the var name in out var for the scope if it is
+# defined, else it will set the outvar to undefined.
+function(zephyr_var_name variable scope out)
+  if(scope STREQUAL "ENV" OR scope STREQUAL "CACHE")
+    if(DEFINED ${scope}{${variable}})
+      set(${out} "$${scope}{${variable}}" PARENT_SCOPE)
+    else()
+      set(${out} PARENT_SCOPE)
+    endif()
+  else()
+    if(DEFINED ${scope}_${variable})
+      set(${out} "${${scope}_${variable}}" PARENT_SCOPE)
+    else()
+      set(${out} PARENT_SCOPE)
```

Comment:
non-blocking: `unset(${out} PARENT_SCOPE)` would be more readable IMO, but I see that `set(foo)` is used elsewhere too.

Reply:
In this particular case, where it is a function and the function sets the value of the return variable referenced by `out`.
Then I personally think it is cleaner to indicate that it sets the value of the return var to nothing (hence clears it), than to unset a return variable.

But I agree, it can be seen both ways.